src/langserver/debugger/mod.rs

```diff
@@ -57,37 +66,49 @@ pub fn start_server(dreamseeker_exe: String, db: DebugDatabaseBuilder) -> std::i
         .spawn(move || {
             let (stream, _) = listener.accept().unwrap();
             drop(listener);
-
+            let env = dm::detect_environment_default()
+                .expect("error detecting .dme")
+                .expect("no .dme found");
+            let ctx = dm::Context::default();
+            ctx.autodetect_config(&env);
```

Comment:
The `dm::Context` or config should be passed in from the language server here, rather than going through `detect_environment`.

Reply:
I'm not too sure on a good way to do this. It seems `struct Engine` does have a context (I'd noticed this but didn't realise it actually loaded the config).

Perhaps instead of passing the context to Debugger:new I can just pass the `engine` config's value.

Reply:
Ok that's done. Hopefully it's fine.

Reply:
Looks good.